.github/workflows/release.yml

```diff
@@ -32,15 +32,21 @@ jobs:
             echo "runs_tests=$(if [ '${{ inputs.skip_tests }}' = 'true' ]; then echo 'false'; else echo 'true'; fi)"
           } >> "$GITHUB_OUTPUT"
 
-  validate-version-input:
+  validate-inputs:
     runs-on: ubuntu-latest
     steps:
       - name: Validation of version format
         run: |
           echo "${{ inputs.version_number }}" | grep -P '^v\d+\.\d+\.\d+(-pre[A-Za-z0-9-]*)?$'
+      - name: Checkout
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633
+        with:
+          ref: ${{ inputs.use_existing_tag == 'true' && inputs.version_number || 'master' }}
+      - name: Check for Upgrade Guide
+        run: './scripts/check-upgrade-guide-exists.sh ${{inputs.version_number}}'
 
   update-examples-reference-in-docs:
-    needs: [ release-config, validate-version-input ]
+    needs: [ release-config, validate-inputs ]
```

Comment:
knit: similary to validate-inputs i don't know if it would be better to have an update job with both references in doc and changelog header.   or maybe it's better as it is now

Reply:
Since the commits are being done by calling an external workflow we do not have the possibility of having both in a single job